packages/video_player/video_player_avfoundation/example/ios/RunnerTests/VideoPlayerTests.m

```diff
@@ -439,4 +546,22 @@ - (void)validateTransformFixForOrientation:(UIImageOrientation)orientation {
   XCTAssertEqual(t.ty, expectY);
 }
 
+- (void)waitForCondition:(BOOL (^)(void))condition withTimeout:(NSTimeInterval)timeout {
```

Comment:
i think you can just use NSPredicate and expectation for this kind of stuff

Reply:
Great idea, I've refactored the code ✅